source/codegen/templates/service_helpers.mako

```diff
@@ -719,30 +713,41 @@ ${initialize_hardcoded_parameter(parameter)}
 
 
 ## Handles populating the response message after calling the driver API.
-<%def name="populate_response(function_data, output_parameters, indent_level=0, init_method=False)">\
+<%def name="populate_response(function_data, parameters, indent_level=0, init_method=False)">\
 <%
+  output_parameters = [p for p in parameters if common_helpers.is_output_parameter(p)]
   get_last_error_outputs = service_helpers.get_last_error_output_params(output_parameters)
   normal_outputs = [p for p in output_parameters if not p in get_last_error_outputs]
 %>\
 <%block filter="common_helpers.indent(indent_level)">\
+${populate_error_check(function_data, parameters, indent_level)}
       response->set_status(status);
 %if output_parameters:
-      if (status_ok(status)) {
 ${set_response_values(normal_outputs, init_method)}\
-      }
-%   if any(get_last_error_outputs):
-      else {
-%     for get_last_error_output in get_last_error_outputs:
+%endif
+</%block>\
+</%def>
+
+## Handles populating the response message after calling the driver API.
+<%def name="populate_error_check(function_data, parameters, indent_level=0)">\
+<%
+  config = data['config']
+  input_parameters = [p for p in parameters if common_helpers.is_input_parameter(p)]
+  resource_handle_types = service_helpers.get_resource_handle_types(config)
+%>\
+<%block filter="common_helpers.indent(indent_level)">\
 <%
-  get_last_error_output_name = common_helpers.get_grpc_field_name(get_last_error_output)
-  get_last_error_method_name = get_last_error_output["get_last_error"]
+  session = 0
+  handle_type = resource_handle_types[0]
+  for parameter in input_parameters:
+    if parameter['type'] in resource_handle_types:
+      session = common_helpers.get_cpp_local_name(parameter)
+      handle_type = parameter['type']
+      break
```

Comment:
I actually don't think that's desirable. Some methods have no session handle parameters. And in fact this would incorrectly assert for all Init methods, because this is looking only at the `in` parameters. In that case we just pick something and that should be good enough.